src/MenuItem.php

```diff
@@ -172,7 +174,7 @@ public function __get($field)
         } else {
             $page = $this->Page();
 
-            if ($page instanceof DataObject) {
+            if ($page->ID != 0 && $page instanceof DataObject) {
```

Comment:
🤔. This is a tough one. I have a few comments here:

1. If we're going with this solution, then could we use `$page->exists()` here?
2. The original logic feels like it's trying to replicate the role of `ViewableData::$failover`, so should we really be factoring to use that pattern instead?
3. The symptom here is a Fluent concern, so is it safe to add opinions without being able to prove them internally to this module?